### DISCUSSION TITLE
Invalid route "/*__private__axum_fallback": insertion failed due to conflict with previously registered route

### DISCUSSION THREAD

**Discussion by alf5** [tokio-rs/axum#2701] [General]
I would like to structure my app with a "pages" router and an "api" router.
they come from seperate modules - easy enough, all good.

Problem is that I wish to add a "ServeDir" as a fallback handler for "/".
I get the panic error `Invalid route "/*__private__axum_fallback": insertion failed due to conflict with previously registered route...:` if I add it to  either the top level Router or the "pages" Router.

Top level router:
```rust
  let app = Router::new()
    .nest("/", pages::router()) // <-- having "nest" throws the panic
    .nest("/api", api::router())
    .fallback_service(ServeDir::new("public"));
```

If I swap the "nest" for a "route" handler it all works.
i.e.
```rust
  let app = Router::new()
    .route("/", get(|| async { "Hello, World!" })) //<-- no conflict, all good!
    .nest("/app", pages::router())
    .nest("/api", api::router())
    .fallback_service(ServeDir::new("public"));
```

As stated, I have also tried putting the `fallback_service` call at the end of the "pages" router but it still panics with similar error (same error but the reported route == "/").

I was wondering what I could be doing wrong?
I am not sure if this is a bug or not.

p.s. the closest I have [read is this discussion](https://github.com/tokio-rs/axum/discussions/2012), and yes, that looks to be the same cause but the solution to move the `fallback_service` call does not solve my situation.

**Comment by jplatte** [comment]
You can use `Router::merge` instead of nesting at `/`.

**Comment by alf5** [comment]
@jplatte - that worked thank you.
